javascript2-week2/quizApp2.js

```diff
@@ -0,0 +1,184 @@
+document.addEventListener('DOMContentLoaded', () => {
+    const quizForm = document.getElementById('quizForm');
+    const questionsList = document.getElementById('questionsList');
+    const filter = document.getElementById('filter');
+    let quizQuestions = [];
+
+    quizForm.addEventListener('submit', (e) => {
+        e.preventDefault();
+        if (validateForm()) {
+            addQuestion();
+        }
+    });
+
+    document.getElementById('randomize').addEventListener('click', randomizeOptions);
+    document.getElementById('correctAnswer').addEventListener('change', updateOptionColors);
+
+    filter.addEventListener('input', filterQuestions);
+
+    function validateForm() {
+        const question = document.getElementById('question').value;
+        const options = Array.from(document.getElementsByClassName('option'));
```

Comment:
Here you can also use `querySelectorAll`. (Read about it here)[https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelectorAll]